src/ImageSharp/Formats/Gif/Sections/GifXmpApplicationExtension.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace SixLabors.ImageSharp.Formats.Gif
+{
+    internal readonly struct GifXmpApplicationExtension : IGifExtension
+    {
+        public GifXmpApplicationExtension(byte[] data) => this.Data = data;
+
+        public byte Label => GifConstants.ApplicationExtensionLabel;
+
+        public int ContentLength => this.Data.Length + 269; // 12 + Data Length + 1 + 256
+
+        /// <summary>
+        /// Gets the raw Data.
+        /// </summary>
+        public byte[] Data { get; }
+
+        /// <summary>
+        /// Reads the XMP metadata from the specified stream.
+        /// </summary>
+        /// <param name="stream">The stream to read from.</param>
+        /// <returns>The XMP metadata</returns>
+        /// <exception cref="ImageFormatException">Thrown if the XMP block is not properly terminated.</exception>
+        public static GifXmpApplicationExtension Read(Stream stream)
+        {
+            // Read data in blocks, until an \0 character is encountered.
+            // We overshoot, indicated by the terminatorIndex variable.
+            const int bufferSize = 256;
+            var list = new List<byte[]>();
```

Comment:
This is quite allocation heavy. Would it make sense to build this around [ReadOnlySequence<T>](https://docs.microsoft.com/en-us/dotnet/standard/io/buffers#readonlysequencet) and / or [Pipelines](https://docs.microsoft.com/en-us/dotnet/standard/io/pipelines)?

Of course, functionality can be brought in and optimized later on.

Reply:
I’d like to revisit a lot of stuff for V3 where we can trim framework targets.